channeldb/payment_control.go

```diff
@@ -530,16 +534,11 @@ func (p *PaymentControl) Fail(paymentHash lntypes.Hash,
 		// lets the last attempt to fail with a terminal write its
 		// failure to the PaymentControl without synchronizing with
 		// other attempts.
-		paymentStatus, err := fetchPaymentStatus(bucket)
+		_, err = fetchPaymentStatus(bucket)
```

Comment:
You are no longer setting `updateErr` and returning `nil` in the `ErrPaymentNotInitiated` case. Is this correct in the context of a `Batch`?

Reply:
interesting. I put it back as it's not meant to change in this PR, but do have a question. Whenever the `updateErr` is set, the `payment` is always nil. Why don't we return the `updateErr` instead? Is it because inside a `Batch`, multiple txes are happening, and an error from a specific tx can halt the whole batching?

Reply:
https://github.com/lightningnetwork/lnd/pull/7175#discussion_r1099116811